KubeArmor/core/karmorprobedata.go

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 Authors of KubeArmor
+
+package core
+
+import (
+	"encoding/json"
+	"log"
+	"os"
+
+	cfg "github.com/kubearmor/KubeArmor/KubeArmor/config"
+)
+
+type KarmorData struct {
+	OSImage             string
+	KernelVersion       string
+	KubeletVersion      string
+	ContainerRuntime    string
+	ActiveLSM           string
+	KernelHeaderPresent bool
+	HostSecurity        bool
+	ContainerSecurity   bool
+	KubeArmorPosture    KubeArmorPostures
+}
+
+type KubeArmorPostures struct {
+	DefaultFilePosture             string
+	DefaultNetworkPosture          string
+	DefaultCapabilitiesPosture     string
+	HostDefaultFilePosture         string
+	HostDefaultNetworkPosture      string
+	HostDefaultCapabilitiesPosture string
+}
+
+func SetKarmorData(dm *KubeArmorDaemon) KarmorData {
```

Comment:
```suggestion
func (dm *KubeArmorDaemon) SetKarmorData {
```

Let's add this as a struct method and not pass the address to the daemon structure here.